packages/build/src/telemetry/main.js

```diff
@@ -65,7 +67,9 @@ const getPayload = function ({ status, commandsCount, netlifyConfig, durationNs,
       status,
       steps: commandsCount,
       buildVersion: version,
-      nodeVersion: nodeVersion.replace('v', ''),
+      // We're passing the NODE_VESION set by the buildbot which will run the `build.command` and
+      // the `package.json`/locally defined plugins
+      nodeVersion: childEnv.NODE_VERSION,
```

Comment:
`NODE_VERSION` does not always match the version run in the build command nor plugins. 
This function is used to retrieve the user Node.js version used in those: https://github.com/netlify/build/blob/fbc235e407ac100957a53fe5a151c02bde58bb7f/packages/build/src/plugins/node_version.js#L22

At the moment, that logic is computed a little later in the main flow, at https://github.com/netlify/build/blob/fbc235e407ac100957a53fe5a151c02bde58bb7f/packages/build/src/core/main.js#L327

However, it is probably possible to move `await getUserNodeVersion(nodePath)` to an earlier step in the code flow. For example, to do in parallel with `package.json` retrieval (https://github.com/netlify/build/blob/fbc235e407ac100957a53fe5a151c02bde58bb7f/packages/build/src/core/config.js#L67).

What do you think?

Reply:
> NODE_VERSION does not always match the version run in the build command nor plugins.

Are there edge cases not covered by the `build-image` install script? I saw that particular code, however I was just reluctant to make another process execution just to extract the node version 😕 If that's the only option though I'll look into it 👍

Reply:
Looking at how `NODE_VERSION` is set in `build-image` and how `--user-node-path` is set in `buildbot`, it seems like those two should always be identical, although they use different logic. 

The `userNodeVersion` variable (and related logic, based on `--user-node-path`) is what's used to determine the Node.js version in plugins and the build command. Would using the same logic (by re-using that variable) provide with stronger guarantee we are logging the Node.js version used in plugins and the build command, even if we were to change the `build-image` script?

Also, `NODE_VERSION` is quite ambiguous because its meaning changes through the build: 
  - First, it is the environment variable set by users to customize the Node.js version
  - Then, it becomes the resolved Node.js version to use in build command and plugins

Based on this, I am wondering if it might be more cautious not to rely on it too much, in order to limit the extent of that ambiguity in the codebase?

That being said, both options should work for the purpose of telemetry :+1:

Reply:
> Based on this, I am wondering if it might be more cautious not to rely on it too much, in order to limit the extent of that ambiguity in the codebase?

💯, you're right, we should probably avoid relying that much on undocumented behaviour that can be changed altogether without us noticing.

I'll look into re-using that logic, probably break it into two separate PRs. Will keep this open to handle the telemetry specifc work.

Reply:
One small detail for reference about the logic in

https://github.com/netlify/build/blob/fbc235e407ac100957a53fe5a151c02bde58bb7f/packages/build/src/plugins/node_version.js#L22

Which should be unrelated, but is worth noting. That logic looks a little more complex than it really is. It just does:
  - If there is a `--node-path`, then run `node --version` with it
  - Otherwise, use `process.version`

The bit about parsing `--node-path` with a RegExp is just a performance optimization to avoid having to execute a process, but it should return the same result.

Reply:
This has been addressed in https://github.com/netlify/build/pull/2634/commits/ed13deb639934f4602dac0c13b1a40860355f0e7 @ehmicky let me know what you think 👍